conftest.py

```diff
@@ -79,6 +79,12 @@ def skipif(items, whole_module=False):
            isinstance(configuration['platform'], Cpu64):
             skipit = "`icx+cpu64` won't work with this test"
             break
+        # Skip if icx or advisor are not available
+        if i == 'noadvisor' and \
+           not isinstance(configuration['compiler'], IntelCompiler) or \
```

Comment:
this is wrong again, I think, because of and/or operator precedence...

The fact we got this wrong multiple times and nothing breaks on CI suggests it's not very useful maybe